src/IIIFPresentation/BackgroundHandler/BatchCompletion/BatchCompletionMessageHandler.cs

```diff
@@ -131,9 +131,12 @@ await iiifS3.SaveIIIFToS3(mergedManifest, batch.Manifest!, $"{dlcsSettings.Prese
 
     private void CompleteBatch(Batch batch, DateTime finished)
     {
-        batch.Processed = DateTime.UtcNow;
+        var processed = DateTime.UtcNow;
+        
+        batch.Processed = processed;
         batch.Finished = finished;
         batch.Status = BatchStatus.Completed;
+        batch.Manifest!.LastProcessed = processed;
```

Comment:
It looks like batch 1 of n could mark the manifest as `LastProcessed` here? Rather than only the final batch doing so?